scenario/mocking.py

```diff
@@ -302,6 +303,19 @@ def secret_add(
         self._state.secrets.append(secret)
         return id
 
+    def _check_can_manage_secret(
+        self,
+        secret: "Secret",
+    ):
+        # FIXME: match real tracebacks
+        if secret.owner is None:
+            raise SecretNotFoundError("this secret is not owned by this unit/app")
+        if secret.owner == "app" and not self.is_leader():
+            raise SecretNotFoundError(
```

Comment:
This is the exception when doing `self.model.get_secret(id=id).get_info()` from a unit that's not the leader:

```
unit-testing-secrets-1: 20:04:11 ERROR unit.testing-secrets/1.juju-log Uncaught exception while in charm code:
Traceback (most recent call last):
  File "/var/lib/juju/agents/unit-testing-secrets-1/charm/venv/ops/model.py", line 2946, in _run
    result = subprocess.run(args, **kwargs)  # type: ignore
  File "/usr/lib/python3.10/subprocess.py", line 526, in run
    raise CalledProcessError(retcode, process.args,
subprocess.CalledProcessError: Command '('/var/lib/juju/tools/unit-testing-secrets-1/secret-get', 'secret://085db680-df30-4f13-82d7-eae3392be3d4/clffid0eq7ms77ta2kbg', '--format=json')' returned non-zero exit status 1.

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/var/lib/juju/agents/unit-testing-secrets-1/charm/./src/charm.py", line 54, in <module>
    ops.main(TestingSecretsCharm)  # type: ignore
  File "/var/lib/juju/agents/unit-testing-secrets-1/charm/venv/ops/main.py", line 451, in __call__
    return main(charm_class, use_juju_for_storage=use_juju_for_storage)
  File "/var/lib/juju/agents/unit-testing-secrets-1/charm/venv/ops/main.py", line 436, in main
    _emit_charm_event(charm, dispatcher.event_name)
  File "/var/lib/juju/agents/unit-testing-secrets-1/charm/venv/ops/main.py", line 144, in _emit_charm_event
    event_to_emit.emit(*args, **kwargs)
  File "/var/lib/juju/agents/unit-testing-secrets-1/charm/venv/ops/framework.py", line 340, in emit
    framework._emit(event)
  File "/var/lib/juju/agents/unit-testing-secrets-1/charm/venv/ops/framework.py", line 842, in _emit
    self._reemit(event_path)
  File "/var/lib/juju/agents/unit-testing-secrets-1/charm/venv/ops/framework.py", line 931, in _reemit
    custom_handler(event)
  File "/var/lib/juju/agents/unit-testing-secrets-1/charm/./src/charm.py", line 39, in _on_get_unit_secret_action
    event.set_results({"password": self.model.get_secret(id=id).get_info()})
  File "/var/lib/juju/agents/unit-testing-secrets-1/charm/venv/ops/model.py", line 276, in get_secret
    content = self._backend.secret_get(id=id, label=label)
  File "/var/lib/juju/agents/unit-testing-secrets-1/charm/venv/ops/model.py", line 3302, in secret_get
    result = self._run('secret-get', *args, return_output=True, use_json=True)
  File "/var/lib/juju/agents/unit-testing-secrets-1/charm/venv/ops/model.py", line 2948, in _run
    raise ModelError(e.stderr) from e
ops.model.ModelError: ERROR permission denied
```